src/windows_registry.rs

```diff
@@ -218,6 +205,48 @@ mod impl_ {
         }
     }
 
+    /// Checks to see if the `VSCMD_ARG_TGT_ARCH` environment variable matches the
+    /// given target's arch. Returns `None` if the variable does not exist.
+    #[cfg(windows)]
+    fn is_vscmd_target(target: &str) -> Option<bool> {
+        let vscmd_arch = env::var("VSCMD_ARG_TGT_ARCH").ok()?;
+        // Convert the Rust target arch to its VS arch equivalent.
+        let arch = match target.split("-").next() {
+            Some("x86_64") => "x64",
+            Some("aarch64") => "arm64",
+            Some("i686") | Some("i586") => "x86",
+            Some("thumbv7a") => "arm",
+            // An unrecognized arch.
+            _ => return Some(false),
+        };
+        Some(vscmd_arch == arch)
+    }
+
+    /// Attempt to find the tool using environment variables set by vcvars.
+    pub fn find_msvc_environment(target: &str, tool: &str) -> Option<Tool> {
+        // Early return if the environment doesn't contain a VC install.
+        if env::var_os("VCINSTALLDIR").is_none() {
+            return None;
+        }
```

Comment:
I think this statement isn't necessary due to the `?` on the one below?

Reply:
The one below is slightly different in that it's `VS*` instead of `VC*`. I was trying to account for the possiblity that Visual Studio (VS) is installed but the Visual C (VC) component isn't.

Reply:
Oh oops missed that, don't mind me!